include/cocaine/format/ptr.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "cocaine/format/base.hpp"
+
+#include <memory>
+namespace cocaine {
+
+template<class T>
+struct ptr_display {
+    using value_type = T;
+    using underlying_type = typename pristine<decltype(*std::declval<T>())>::type;
+
+    static
+    auto
+    apply(std::ostream& stream, const value_type& value) -> std::ostream& {
+        if(value) {
+            display<underlying_type>::apply(stream, *value);
+        } else {
+            stream << "none";
```

Comment:
I'd rather print `nullptr`, because there is significant difference between optional and null-pointer cases.